packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -59,6 +69,7 @@ type State = {
   connections: ConnectionInfo[];
   oidcDeviceAuthVerificationUrl: string | null;
   oidcDeviceAuthUserCode: string | null;
+  connectionMergeInfos: Record<string, RecursivePartial<ConnectionInfo>>;
```

Comment:
Would it help to leave a comment here? The name `connectionMergeInfos`, at least for me, doesn't say much about what it is or used for. Can't think of a better name myself.
Maybe:
```suggestion
  // Additional connection information that is merged with the connection info
  // when connecting. This is useful for instances like OIDC sessions where we
  // have a setting on the system for storing credentials.
  // When the setting is on this `connectionMergeInfos` would have the session
  // credential information and merge it before connecting.
  connectionMergeInfos: Record<string, RecursivePartial<ConnectionInfo>>;
```

Reply:
Yeah, that makes perfect sense to me :+1: